python/dgl/sampling/neighbor.py

```diff
@@ -129,18 +128,24 @@ def sample_neighbors(g, nodes, fanout, edge_dir='in', prob=None, replace=False,
         else:
             nodes_all_types.append(nd.array([], ctx=nd.cpu()))
 
-    if not isinstance(fanout, dict):
-        fanout_array = [int(fanout)] * len(g.etypes)
+    if isinstance(fanout, nd.NDArray):
+        fanout_array = fanout
     else:
-        if len(fanout) != len(g.etypes):
-            raise DGLError('Fan-out must be specified for each edge type '
-                           'if a dict is provided.')
-        fanout_array = [None] * len(g.etypes)
-        for etype, value in fanout.items():
-            fanout_array[g.get_etype_id(etype)] = value
-    fanout_array = F.to_dgl_nd(F.tensor(fanout_array, dtype=F.int64))
-
-    if prob is None:
+        if not isinstance(fanout, dict):
+            fanout_array = [int(fanout)] * len(g.etypes)
```

Comment:
same issue of `None` type `fanout` argument as above

Reply:
The Doc String says:
```
fanout : int or dict[etype, int]
```

It looks like calling code should convert `None`s to `-1`.
```
If -1 is given for a single edge type, all the neighboring edges with that edge
        type will be selected.
```

Reply:
I was confused by the docstring. Fan-out value of each edge type cannot be None, but the whole `fanout` argument could be None. I think we should improve the docstring later. No more question on this.